### PR TITLE
Dataops 606 methylseq control references

### PR DESCRIPTION
**Pull request by b97pla**
This PR downloads the pUC19 and lambda references used for control spike-ins in methylseq libraries.

It also refactors the pipeline-specific deployment mechanism.

### CODE REVIEW

**Review by aanil — APPROVED** [review]
👍

File: roles/nf-core/tasks/methylseq_include.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+- name: create directory for {{ pipeline }}  resources
+  file:
+    path: "{{ ngi_containers }}/{{ pipeline }}"
+    state: directory
+
+# the eutils service will only tolerate 3 requests / second so make sure not to exceed this
+- name: download FASTA references
+  get_url:
```
Comment (matrulda) [inline roles/nf-core/tasks/methylseq_include.yml:9]: This module will automatically check if the file is already downloaded, right?

File: roles/nf-core/tasks/pipeline.yml [hunk 2]
```diff
@@ -58,28 +58,18 @@
...
-    register: silva_stat
-
-  - name: download SILVA classifier data
-    get_url:
-      url: "{{ silva_base_url }}/{{ silva_zip }}"
-      dest: "{{ ngi_containers }}/{{ pipeline }}/{{ silva_zip }}"
-    when: not silva_stat.stat.exists
-  when: pipeline == "ampliseq"
+- name: execute pipeline-specific includes
+  include_tasks: "{{ pipeline }}_include.yml"
```
Comment (matrulda) [inline roles/nf-core/tasks/pipeline.yml:71]: Nice solution :+1:

**Review by matrulda — APPROVED** [review]
This looks good to me :+1:  Just a minor comment.

File: roles/nf-core/tasks/methylseq_include.yml (hunk 1 above)
Comment (b97pla) [reply roles/nf-core/tasks/methylseq_include.yml:9]: yes, from the docs: 
```
if a checksum is passed to this parameter, and the file exist under the [dest](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html#ansible-collections-ansible-builtin-get-url-module-parameter-dest) location, the destination_checksum would be calculated, and if checksum equals destination_checksum, the file download would be skipped (unless [force](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html#ansible-collections-ansible-builtin-get-url-module-parameter-force) is true). If the checksum does not equal destination_checksum, the destination file is deleted.
```

Comment (matrulda) [reply roles/nf-core/tasks/methylseq_include.yml:9]: Awesome!

**Pull request merged**